File: app/src/main/java/io/github/jonzarate/bloomwild/model/data/ProductAttributes.java
package io.github.jonzarate.bloomwild.model.data;

import com.google.gson.annotations.SerializedName;

public class ProductAttributes {
    @SerializedName("name")
    private String name;

    @SerializedName("collection_name")
    private String collectionName;

    @SerializedName("currency")
    private String currency;

    @SerializedName("price_data")
    private ProductPrice[] prices;
    private String sanitisedPrice;

    @SerializedName("media")
    private ProductMedia[] media;
    private String[] sanitisedMedia;

    private boolean isHighlighted;


    public String getName() {
        return name;
    }

    public String getCollectionName() {
        return collectionName;
    }

    public String getCurrency() {
        return currency;
    }

    public ProductPrice[] getPrices() {
        return prices;
    }

    public String getSanitisedPrice() {
        return sanitisedPrice;
    }

    public void setSanitisedPrice(String sanitisedPrice) {
        this.sanitisedPrice = sanitisedPrice;
    }

    public ProductMedia[] getMedia() {
        return media;
    }

    public String[] getSanitisedMedia() {
        return sanitisedMedia;
    }

    public void setSanitisedMedia(String[] sanitisedMedia) {
        this.sanitisedMedia = sanitisedMedia;
    }

    public boolean isHighlighted() {
        return isHighlighted;
    }

    public void setHighlighted(boolean highlighted) {
        this.isHighlighted = highlighted;
    }
}

File: app/src/main/java/io/github/jonzarate/bloomwild/model/data/ProductAttributes.kt
package io.github.jonzarate.bloomwild.model.data

import com.google.gson.annotations.SerializedName

data class ProductAttributes (
    @SerializedName("name") val name: String,
    @SerializedName("collection_name") val collectionName: String,
    @SerializedName("currency") val currency: String,
    @SerializedName("price_data") val prices: List<ProductPrice>,
    @SerializedName("media") val media: List<ProductMedia>,
    var sanitisedPrice: String? = null,
    var sanitisedMedia: List<String>? = null
) {
    var isHighlighted = false
}